chronicler/events/core/git.py

```diff
@@ -143,3 +181,116 @@ def _process_action(self, source, time, event_uuid, action, file_data):
+                                                   committers)
+        events.extend(identity_events)
+
+        for trailer, event_type in COMMIT_TRAILERS.items():
+            signers = raw_item["data"].get(trailer, [])
+            identity_events = self._process_identities(parent_event['source'],
+                                                       parent_event['time'],
+                                                       parent_event['id'],
+                                                       event_type,
+                                                       signers)
+            events.extend(identity_events)
+
+        return events
+
+    def _process_identities(
+        self,
+        source: str,
+        time: str,
+        event_uuid: str,
+        event_type: str,
+        raw_identities: list[str]
+    ) -> Generator[CloudEvent, None, None]:
+        """Obtain identity events from a list of identities.
+
+        :param source: data source of the event
+        :param time: time of the event
+        :param event_uuid: UUID of the parent event
+        :param event_type: type of the identity event
+        :param raw_identities: list of strings with the identities information
+
+        :returns: generator of CloudEvent with the identity information
+        """
+        for raw_identity in raw_identities:
+            try:
+                identity = self._parse_identity(raw_identity)
+                identity_id = generate_uuid(source="git",
+                                            email=identity.email,
+                                            name=identity.name,
+                                            username=identity.username)
+            except ValueError as e:
+                logger.warning(f"Cannot generate UUID for identity '{raw_identity}' "
+                               f"in event '{event_uuid}': {e}. Skipping.")
+                continue
+
+            role = event_type.split('.')[-1]
+            event_id = uuid(event_uuid, role, identity_id)
+
+            data = {
+                "source": "git",
+                "name": identity.name,
+                "username": identity.username,
+                "email": identity.email,
+                "role": role,
+                "uuid": identity_id,
+            }
+
+            attributes = {
+                "id": event_id,
+                "linked_event": event_uuid,
+                "type": event_type,
+                "source": source,
+                "time": time,
+            }
+
+            yield CloudEvent(attributes, data)
+
+    @staticmethod
+    def _parse_authors(authors: str) -> list[str]:
+        """Parse a list of authors from a string."""
+
+        m = AUTHOR_P2P_REGEX.match(authors)
+        if m:
+            authors = m.group("first_authors").split(",")
+            authors = [author.strip() for author in authors]
+            authors += [m.group("last_author")]
+            authors += [f"<{m.group('email')}>"]
```

Comment:
I think the email address should be shared by all the authors. With the patterns we have seen, it's impossible to know whether it belongs to the first author, second, etc. Sometimes, they even share an email address. I think the safest thing is to assign it to all the identities at the same time.

If these events are handled later, the component that process them show be aware that there's more than one author are sharing an account.

Reply:
That's normally the case and that could happen, the problem is we don't know whose email is it from the list of identities listed on it. However, I understand what you need let's keep the email as a separate individual.